skylark/skydoc.bzl

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Skylark rule for skydoc: a documentation generator tool written in Java."""
+
+load("@bazel_skylib//:skylark_library.bzl", "SkylarkLibraryInfo")
+
+def _skydoc_impl(ctx):
+    """Implementation of the skydoc rule."""
+    out_file = ctx.outputs.out
+    input_files = depset(order = "postorder", direct = [ctx.files.target_file[0]], transitive = [
```

Comment:
Does the order of the input files matter to skydoc? If not, use `default` order (i.e., omit the `order` arg entirely) for maximum compatibility with other depsets (though in this case the only inputs should be from `SkylarkLibraryInfo`).

Reply:
It doesn't care about the order of the input files, no (could any action care, actually? This isn't the order of the args, but the order that input files are included, which seems like it wouldn't matter on actual action execution. I'm curious how this would matter, though I'm willing to believe it somehow might :))